experimental/dds/tree/src/id-compressor/IdCompressor.ts

```diff
@@ -668,7 +603,23 @@ export class IdCompressor {
 						'The number of allocated final IDs must not exceed the JS maximum safe integer.'
 					);
 					this.checkClusterForCollision(currentCluster);
+					if (isLocal) {
+						// Example:
+						// Ids generated so far:   -1  1  2 -4 -5  <-- note positive numbers are eager finals
+						//         Cluster:      [  0  1  2 ]
+						// ~ finalizing happens, causing expansion ~
+						//        Cluster:       [  0  1  2  3  4  5 ]
+						// corresponding locals:   -1       -4
+						//      lastFinalizedLocalId^              ^newLastFinalizedLocalId = -6
+						//                  overflow = 2:    ----
+						//                       localIdPivot^
+						//                    lastFinalizedFinal^
+						const lastFinalizedFinal = (currentBaseFinalId + currentCluster.count - 1) as FinalCompressedId;
+						const finalPivot = (lastFinalizedFinal - overflow + 1) as FinalCompressedId;
+						this.sessionIdNormalizer.addFinalIds(finalPivot, lastFinalizedFinal, currentCluster);
+					}
 				}
+				remainingCount = 0;
```

Comment:
Don't think this assignment is necessary since it's already being reassigned to 0 above and isn't being touched in any code in between

Reply:
agreed